src/main/java/com/exactpro/th2/conn/ConnectivityBatch.java

```diff
@@ -57,42 +58,54 @@ public RawMessageBatch convertToProtoRawBatch() {
                 .build();
     }
 
-    public String getSessionAlias() {
-        return sessionAlias;
+    public String getBookName() {
+        return bookName;
     }
 
-    public long getSequence() {
-        return sequence;
+    public String getSessionAlias() {
+        return sessionAlias;
     }
 
     public Direction getDirection() {
         return direction;
     }
 
+    public long getSequence() {
+        return sequence;
+    }
+
     public List<ConnectivityMessage> getMessages() {
         return connectivityMessages;
     }
 
-    private static void checkMessages(List<ConnectivityMessage> iMessages, String sessionAlias, Direction direction) {
+    private static void checkMessages(List<ConnectivityMessage> iMessages, String bookName, String sessionAlias, Direction direction) {
         if (iMessages.isEmpty()) {
             throw new IllegalArgumentException("List can't be empty");
         }
 
-        if (!iMessages.stream()
-                .allMatch(iMessage -> Objects.equals(sessionAlias, iMessage.getSessionAlias())
-                        && direction == iMessage.getDirection())) {
-            throw new IllegalArgumentException("List " + iMessages + " has elemnts with incorrect metadata, expected session alias '"+ sessionAlias +"' direction '" + direction + '\'');
+        if (iMessages.stream()
+                .anyMatch(iMessage -> !Objects.equals(bookName, iMessage.getBookName())
+                        || !Objects.equals(sessionAlias, iMessage.getSessionAlias())
+                        || direction != iMessage.getDirection())
```

Comment:
Maybe return to the version with one negation

Reply:
I thought that `.anyMatch()` is more readable: if there is any inappropriate message we'll throw `IllegalArgumentException` and we don't need to check all.
Please let me know if I even should return `!.allMatch()`.

Reply:
done